infra/cifuzz/run_fuzzers.py

```diff
@@ -27,61 +27,167 @@
+
+
 def run_fuzzers(config):  # pylint: disable=too-many-locals
   """Runs fuzzers for a specific OSS-Fuzz project.
 
   Args:
-    fuzz_seconds: The total time allotted for fuzzing.
-    workspace: The location in a shared volume to store a git repo and build
-      artifacts.
-    project_name: The name of the relevant OSS-Fuzz project.
-    sanitizer: The sanitizer the fuzzers should be run with.
+    config: A RunFuzzTargetsConfig.
 
   Returns:
-    (True if run was successful, True if bug was found).
+    (True if no (internal) errors fuzzing, True if bug found fuzzing).
   """
-  # Validate inputs.
-  logging.info('Using %s sanitizer.', config.sanitizer)
-
-  out_dir = os.path.join(config.workspace, 'out')
-  artifacts_dir = os.path.join(out_dir, 'artifacts')
-  os.makedirs(artifacts_dir, exist_ok=True)
-
-  if not config.fuzz_seconds or config.fuzz_seconds < 1:
-    logging.error('Fuzz_seconds argument must be greater than 1, but was: %s.',
-                  config.fuzz_seconds)
-    return False, False
-
-  # Get fuzzer information.
-  fuzzer_paths = utils.get_fuzz_targets(out_dir)
-  if not fuzzer_paths:
-    logging.error('No fuzzers were found in out directory: %s.', out_dir)
+  fuzz_target_runner = get_fuzz_target_runner(config)
+  # TODO(metzman): Multiple return bools is confusing. Change to one enum
+  # return value.
+  if not fuzz_target_runner.initialize():
+    # We didn't fuzz at all because of internal (CIFuzz) errors. And we didn't
+    # find any bugs.
     return False, False
 
-  # Run fuzzers for allotted time.
-  total_num_fuzzers = len(fuzzer_paths)
-  fuzzers_left_to_run = total_num_fuzzers
-  min_seconds_per_fuzzer = config.fuzz_seconds // total_num_fuzzers
-  for fuzzer_path in fuzzer_paths:
-    run_seconds = max(config.fuzz_seconds // fuzzers_left_to_run,
-                      min_seconds_per_fuzzer)
-
-    target = fuzz_target.FuzzTarget(fuzzer_path,
-                                    run_seconds,
-                                    out_dir,
-                                    config.project_name,
-                                    sanitizer=config.sanitizer)
-    start_time = time.time()
-    testcase, stacktrace = target.fuzz()
-    config.fuzz_seconds -= (time.time() - start_time)
-    if not testcase or not stacktrace:
-      logging.info('Fuzzer %s, finished running.', target.target_name)
-    else:
-      utils.binary_print(b'Fuzzer %s, detected error:\n%s' %
-                         (target.target_name.encode(), stacktrace))
-      shutil.move(testcase, os.path.join(artifacts_dir, 'test_case'))
-      stack_parser.parse_fuzzer_output(stacktrace, artifacts_dir)
-      return True, True
-    fuzzers_left_to_run -= 1
+  if not fuzz_target_runner.run_fuzz_targets():
+    # We fuzzed successfully, but didn't find any bugs (in the fuzz target).
+    return True, False
 
-  return True, False
+  # We fuzzed successfully and found bug(s) in the fuzz targets.
+  return True, True
```

Comment:
This False, True, 
True, True return get confusing, see if can be refactored in future.

Reply:
Agreed. I want to replace it all with an enum.